pkg/operator/csi/controller/credentials/controller.go

```diff
@@ -0,0 +1,91 @@
+package credentials
+
+import (
+	"context"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	opv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// Controller is a simple controller that maintains a CredentialsRequest static manifest.
+// It uses unstructured.Unstructured as currently there's no API type for this resource.
+type Controller struct {
+	name            string
+	operatorClient  v1helpers.OperatorClient
+	targetNamespace string
+	manifest        []byte
+	dynamicClient   dynamic.Interface
+}
+
+// New returns a Controller.
+func New(
+	name,
+	targetNamespace string,
+	manifest []byte,
+	dynamicClient dynamic.Interface,
+	operatorClient v1helpers.OperatorClient,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &Controller{
+		name:            name,
+		operatorClient:  operatorClient,
+		targetNamespace: targetNamespace,
+		manifest:        manifest,
+		dynamicClient:   dynamicClient,
+	}
+	return factory.New().WithInformers(
+		operatorClient.Informer(),
+	).WithSync(
+		c.sync,
+	).ResyncEvery(
+		time.Minute,
+	).WithSyncDegradedOnError(
+		operatorClient,
+	).ToController(
+		name,
+		recorder.WithComponentSuffix(name),
```

Comment:
I recommend suffixing or prefixing the name so that you're guaranteed of a good name in events and metrics even by notice users

Reply:
Only the event recorder or the controller name as well?

We use the controller name to prefix the conditions, so I'm afraid if I prefix it with `CSIDriverController` and someone sets the controller name as `AWSEBSCSIDriverController` we are going to end up with ackward conditions in the status,